chainer.testing.product raises an error when using unicode literals in python 2

```python
# from __future__ import unicode_literals
import unittest
from chainer import testing


@testing.parameterize(*testing.product({
    'aaa': [1],
}))
class TestNothing(unittest.TestCase):

    def test(self):
        pass
```
When I uncomment the first line of the code above, `pytest test_nothing.py` with Python 2.7.15 raises an error when as below.
```
test_nothing.py:7: in <module>
    'aaa': [1],
chainer/testing/parameterized.py:133: in product
    for names, values in sorted(parameter.items())])
chainer/testing/parameterized.py:102: in _values_to_dicts
    assert isinstance(names, str)
E   AssertionError
```

This issue is causing CI failure for ChainerRL.

--- a/chainer/testing/parameterized.py
+++ b/chainer/testing/parameterized.py
@@ -99,7 +99,7 @@ def parameterize(*params):
 
 
 def _values_to_dicts(names, values):
-    assert isinstance(names, str)
+    assert isinstance(names, six.string_types)
     assert isinstance(values, (tuple, list))
 
     def safe_zip(ns, vs):

